lib/authenticated-rest-api/authenticated-rest-api.ts

```diff
@@ -0,0 +1,272 @@
+            // which we do not use in serverless projects. Remove this from the bundle created
+            // by esbuild to remove unnecessary issues with gcc/g++/python/node which have
+            // no impact as we do not use redis here.
+            "hiredis",
+          ],
+        },
+
+        environment: buildLambdaEnvironment({
+          environment: {
+            PERSONA_CLIENT_NAME: `${apiName}-authoriser`,
+            PERSONA_HOST: props.persona.host,
+            PERSONA_SCHEME: props.persona.scheme,
+            PERSONA_PORT: props.persona.port,
+            PERSONA_OAUTH_ROUTE: props.persona.oauth_route,
+            SCOPE_CONFIG: JSON.stringify(scopeConfig),
+          },
+          timeout: authLambdaTimeout,
+        }),
+
+        awsSdkConnectionReuse: true,
+        runtime: lambda.Runtime.NODEJS_18_X,
+        timeout: authLambdaTimeout,
+        securityGroups: props.securityGroups,
+        vpc: props.vpc,
+        vpcSubnets: props.vpcSubnets,
+      },
+    );
+
+    const authorizer = new apigateway.TokenAuthorizer(
+      this,
+      "lambda-authorizer",
+      {
+        handler: authLambda,
+        resultsCacheTtl: cdk.Duration.seconds(0),
+      },
+    );
+
+    if (props.resourceProps) {
+      for (const resourceProp of props.resourceProps) {
+        this.addResource(apiName, resourceProp, authorizer);
+      }
+    }
+
+    // Add a cloudwatch alarm for the latency of the api - this is all routes within the api
+    const latencyThreshold = props.apiLatencyAlarmThreshold
+      ? props.apiLatencyAlarmThreshold
+      : DEFAULT_API_LATENCY_THRESHOLD;
+    const metricLatency = this.restApi
+      .metricLatency()
+      .with({ statistic: "average", period: cdk.Duration.minutes(1) });
+
+    const routeLatencyAlarm = new cloudwatch.Alarm(
+      this,
+      `${apiName}-latency-alarm`,
+      {
+        alarmName: `${apiName}-latency-alarm`,
+        alarmDescription: `Alarm if latency on api ${props.prefix}${
+          props.name
+        } exceeds ${latencyThreshold.toMilliseconds()} milliseconds`,
+        actionsEnabled: true,
+        metric: metricLatency,
+        evaluationPeriods: 1,
+        threshold: latencyThreshold.toMilliseconds(),
+        comparisonOperator:
+          cloudwatch.ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
+        // Set treatMissingData to IGNORE
+        // Stops alarms with minimal data having false alarms when they transition to this state
+        treatMissingData: cloudwatch.TreatMissingData.IGNORE,
+      },
+    );
+    routeLatencyAlarm.addAlarmAction(this.alarmAction);
+    routeLatencyAlarm.addOkAction(this.alarmAction);
+  }
+
+  addResource(
+    apiName: string,
+    resourceProps: ResourceProps,
+    authorizer: apigateway.IAuthorizer,
+    parent?: apigateway.Resource,
+  ) {
+    let actualParent = this.restApi.root;
+    if (parent) {
+      actualParent = parent;
+    }
```

Comment:
```suggestion
    const actualParent = parent ?? this.restApi.root;
```

If it doesn't change, `const` removes the cognitive load of having to read the code to find that out.